src/limited.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{Body, SizeHint};
+use bytes::Buf;
+use http::HeaderMap;
+use std::error::Error;
+use std::fmt;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A length limited body.
+///
+/// This body will return an error if more than the configured number
+/// of bytes are returned on polling the wrapped body.
+#[derive(Clone, Copy, Debug)]
+pub struct Limited<B> {
+    remaining: usize,
+    inner: B,
+}
+
+impl<B> Limited<B> {
+    /// Create a new `Limited`.
+    pub fn new(inner: B, limit: usize) -> Self {
+        Self {
+            remaining: limit,
+            inner,
+        }
+    }
+}
+
+impl<B> Body for Limited<B>
+where
+    B: Body + Unpin,
```

Comment:
Should use `pin_project` instead and not require `Unpin`? Honestly not sure how much that matters but feels safer and its what we do elsewhere for other bodies.

Reply:
Can make that happen. `pin_project_lite` didn't know how to deal with const generics, so I was a bit more manual here.